packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -15,15 +15,175 @@
+            result(false)
+            return
+        }
+    }
+    
+    func onQuery(args: [String: Any], flutterResult: @escaping FlutterResult) {
+        do {
+            guard let modelName = args["modelName"] as? String else {
+                FlutterDataStoreErrorHandler.prepareError(
+                    flutterResult: flutterResult,
+                    msg: FlutterDataStoreErrorMessage.MALFORMED.rawValue,
+                    errorMap: ["MALFORMED_REQUEST": "modelName was not passed in the arguments." ])
+                return
+            }
+            guard let modelSchema = flutterModelRegistration.modelSchemas[modelName] else {
+                FlutterDataStoreErrorHandler.prepareError(
+                    flutterResult: flutterResult,
+                    msg: FlutterDataStoreErrorMessage.MALFORMED.rawValue,
+                    errorMap: ["MALFORMED_REQUEST": "schema for model \(modelName) is not registered." ])
+                return
+            }
+            let queryPredicates = try QueryPredicateBuilder.fromSerializedMap(serializedMap: args["queryPredicate"] as? [String : Any])
+            let querySortInput = try QuerySortBuilder.fromSerializedList(serializedList: args["querySort"] as? [[String: Any]])
+            let queryPagination = QueryPaginationBuilder.fromSerializedMap(serializedMap: args["queryPagination"] as? [String: Any])
+            try bridge.onQuery(SerializedModel.self,
+                              modelSchema: modelSchema,
+                              where: queryPredicates,
+                              sort: querySortInput,
+                              paginate: queryPagination) { (result) in
+                switch result {
+                case .failure(let error):
+                    print("Query API failed. Error = \(error)")
+                    FlutterDataStoreErrorHandler.handleDataStoreError(error: error,
+                                                                      flutterResult: flutterResult,
+                                                                      msg: FlutterDataStoreErrorMessage.QUERY_FAILED.rawValue)
+                case .success(let res):
+                    let serializedResults = res.map { (queryResult) -> [String: Any] in
+                        return queryResult.toJSON(modelSchema: modelSchema)
+                    }
+                    flutterResult(serializedResults)
+                    return
+                }
+            }
+        } catch {
+            print("Failed to parse query arguments with \(error)")
+            FlutterDataStoreErrorHandler.prepareError(
+                flutterResult: flutterResult,
+                msg: FlutterDataStoreErrorMessage.MALFORMED.rawValue,
+                errorMap: ["UNKNOWN": "\(error.localizedDescription).\nAn unrecognized error has occurred. See logs for details." ])
+            return
+        }
+    }
+    
+    private func createTempPosts() throws {
```

Comment:
We should probably make pivotal tasks to remove these temp post methods.

Reply:
yeah the idea is that configure/save tasks would organically replace these. I'll update the corresponding tasks instead of creating new ones.